ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -340,10 +312,61 @@ def create_random_action(self, n_agents: int) -> np.ndarray:
                         size=(n_agents),
                         dtype=np.int32,
                     )
-                    for i in range(self.action_size)
+                    for i in range(self.discrete_size)
                 ]
             )
-            return action
+        return action
+
+    def validate_action(
```

Comment:
make private if it is not meant for the user to use directly. Alternatively, make this a private utility method.

Reply:
wed be calling a private instance method in environment.py if it was made private. It was made an ActionSpec method by @chriselion 's request.

Reply:
I think adding an underscore is a good idea. This sets the expectation that it's not meant for general consumption and might change in the future (it will probably change in the next PR when we're passing ActionBuffers or some other class instead of raw numpy arrays).

It isn't private (this is python, there's no such thing) so it's fine to keep calling from our own code.